### PR TITLE
Audit and Fix `arrays.md`

### PR DESCRIPTION
**Pull request by lyu4321**
Changes I made for issue #40:
- Fix typos
- Add alt tag/descriptive text for images and links
- Add inter-site link
- Add Frontmatter
- Add syntax highlighting
- For tables, add <thead> and <tbody> tags and changed colspan to colSpan

### CODE REVIEW

**Comment by Yoda-Canada** [comment]
Hey Leyang, you did a good job and fixed a  lot of bugs. I find an issue and share it with you. I checked the arrays.md in the dark model, and an image in the Introduction part has an issue. The word "Value" can't display in the dark model. I suggest you cut this image in the white model and save it in the /static/imag. Then quote this path in your code.

**Comment by Yoda-Canada** [comment]
In addition, you'd better use alt tags for the image. See https://supercooldesign.co.uk/blog/how-to-write-good-alt-text for suggestions. e.g.

**Comment by mkim219** [comment]
Hey @lyu4321. I have reviewed your page. You did really good job to fix the bugs and re-formatted the tables. I found really small things to improve(not a bug).

array

The image does not contain `alt` tag. On my opinion, the image better to have `alt` tag for SEO and accessibility

**Comment by JerryHue** [comment]
Hello, @lyu4321! 

I notice that tables like the following:

image

can be displayed as images instead of raw HTML elements.That way, the table can be resized as an image when displayed on several viewports. Currently, at normal zoom, the table looks like the following on my viewport:

image

Notice that the table has to overflow to be able to show the proper width of its cells, and thus I end up having to scroll left and right to see the table in its entirety.

PR #60 did something along that line of work. You may ask @dbelokon for help, if you'd like!

File: docs/C-Data-Structures/arrays.md [hunk 1]
```diff
@@ -86,13 +100,17 @@ This coding style facilitates modifiability.  If we change the size, we need to
...
+```
 
 For example, to access the first element of grade, we write:
 
-**`grade[0]`**
+```c
+grade[0]
+```
 
 To display all elements of grade, we iterate:
```
Comment (kiennguyenchi) [inline docs/C-Data-Structures/arrays.md:115]: You should add "int i = 0"

File: docs/C-Data-Structures/arrays.md [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
+Programs can process extremely large amounts of data much faster than well-established manual techniques.  Whether this processing is efficient or not depends in large part on how that data is organized.  For example, large collections of data can be organized in [structures](../C-Data-Structures/structures "Structures") if each variable shares the same type with all other variables and the variables are stored contiguously in memory.  Not only can structured data be processed efficiently but the programming of tasks performed on structured data can be simplified considerably.  Instead of coding a separate instruction for each variable, we code the instruction that is common to all variables and apply that instruction in an iteration across the data structure.
```
Comment (kiennguyenchi) [inline docs/C-Data-Structures/arrays.md:19]: Please check, this link gives an error.

File: docs/C-Data-Structures/arrays.md (hunk 1 above)
Comment (lyu4321) [reply docs/C-Data-Structures/arrays.md:115]: Sorry what do you mean? Like grade[int i = 0]? I don't think you need to initialize the index before you use it.

Comment (kiennguyenchi) [reply docs/C-Data-Structures/arrays.md:115]: I mean on line 119, it should be 
```
for (int i = 0; i < NGRADES; i++) {
```

Comment (lyu4321) [reply docs/C-Data-Structures/arrays.md:115]: Oh got it. Nice catch! Thanks :)

File: docs/C-Data-Structures/arrays.md (hunk 2 above)
Comment (lyu4321) [reply docs/C-Data-Structures/arrays.md]: Thanks! I've updated the intra-site links.

**Review by humphd — APPROVED** [review]

**Pull request merged**